### ISSUE TITLE
ClickHouse crash: SerializationSparse::deserializeTextEscaped

### ISSUE THREAD

**Issue by alexsubota** [ClickHouse/ClickHouse#72943]
ClickHouse v.24.10.2.80

````
2024.12.09 11:46:53.087335 [ 815 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2024.12.09 11:46:53.087406 [ 815 ] {} <Fatal> BaseDaemon: (version 24.10.2.80 (official build), build id: 5FEEB214EC39870251E7BE1C5C83408145CE2712, git hash: 96b80057159a167b4c697cf2676afa9319f87f21, architecture: x86_64) (from thread 731) Received signal 11
2024.12.09 11:46:53.087444 [ 815 ] {} <Fatal> BaseDaemon: Signal description: Segmentation fault
2024.12.09 11:46:53.087455 [ 815 ] {} <Fatal> BaseDaemon: Address: 0x77. Access: write. Address not mapped to object.
2024.12.09 11:46:53.087490 [ 815 ] {} <Fatal> BaseDaemon: Stack trace: 0x000000000d2765e1 0x00007f7f2ff66420 0x0000000007a5d93f 0x000000001080c84b 0x0000000012bda08e 0x0000000012a5c1d9 0x0000000012a4feac 0x0000000012a5bc8b 0x0000000012a5be9d 0x0000000012a2bb39 0x00000000129cc3d6 0x00000000129a8b1b 0x00000000129a8827 0x0000000012b336e1 0x000000000d04eb8b 0x000000000d0553a2 0x000000000d04c0bf 0x000000000d052f5a 0x00007f7f2ff5a609 0x00007f7f2fe7f353
2024.12.09 11:46:53.087505 [ 815 ] {} <Fatal> BaseDaemon: ########################################
2024.12.09 11:46:53.087518 [ 815 ] {} <Fatal> BaseDaemon: (version 24.10.2.80 (official build), build id: 5FEEB214EC39870251E7BE1C5C83408145CE2712, git hash: 96b80057159a167b4c697cf2676afa9319f87f21) (from thread 731) (query_id: 23220baa-2a90-4f1b-aa85-8e2388af18cf) (query: 
INSERT INTO table (list_of_columns)
 FORMAT TabSeparated
) Received signal Segmentation fault (11)
2024.12.09 11:46:53.087558 [ 815 ] {} <Fatal> BaseDaemon: Address: 0x77. Access: write. Address not mapped to object.
2024.12.09 11:46:53.087566 [ 815 ] {} <Fatal> BaseDaemon: Stack trace: 0x000000000d2765e1 0x00007f7f2ff66420 0x0000000007a5d93f 0x000000001080c84b 0x0000000012bda08e 0x0000000012a5c1d9 0x0000000012a4feac 0x0000000012a5bc8b 0x0000000012a5be9d 0x0000000012a2bb39 0x00000000129cc3d6 0x00000000129a8b1b 0x00000000129a8827 0x0000000012b336e1 0x000000000d04eb8b 0x000000000d0553a2 0x000000000d04c0bf 0x000000000d052f5a 0x00007f7f2ff5a609 0x00007f7f2fe7f353
2024.12.09 11:46:53.087616 [ 815 ] {} <Fatal> BaseDaemon: 0. signalHandler(int, siginfo_t*, void*) @ 0x000000000d2765e1
2024.12.09 11:46:53.087631 [ 815 ] {} <Fatal> BaseDaemon: 1. ? @ 0x00007f7f2ff66420
2024.12.09 11:46:53.087664 [ 815 ] {} <Fatal> BaseDaemon: 2. DB::SerializationNumber<long>::deserializeText(DB::IColumn&, DB::ReadBuffer&, DB::FormatSettings const&, bool) const @ 0x0000000007a5d93f
2024.12.09 11:46:53.087690 [ 815 ] {} <Fatal> BaseDaemon: 3. DB::SerializationSparse::deserializeTextEscaped(DB::IColumn&, DB::ReadBuffer&, DB::FormatSettings const&) const @ 0x000000001080c84b
2024.12.09 11:46:53.087719 [ 815 ] {} <Fatal> BaseDaemon: 4. DB::TabSeparatedFormatReader::readField(DB::IColumn&, std::shared_ptr<DB::IDataType const> const&, std::shared_ptr<DB::ISerialization const> const&, bool, String const&) @ 0x0000000012bda08e
2024.12.09 11:46:53.087747 [ 815 ] {} <Fatal> BaseDaemon: 5. DB::RowInputFormatWithDiagnosticInfo::deserializeFieldAndPrintDiagnosticInfo(String const&, std::shared_ptr<DB::IDataType const> const&, DB::IColumn&, DB::WriteBuffer&, unsigned long) @ 0x0000000012a5c1d9
2024.12.09 11:46:53.087772 [ 815 ] {} <Fatal> BaseDaemon: 6. DB::RowInputFormatWithNamesAndTypes<DB::TabSeparatedFormatReader>::parseRowAndPrintDiagnosticInfo(std::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn>>>&, DB::WriteBuffer&) @ 0x0000000012a4feac
2024.12.09 11:46:53.087803 [ 815 ] {} <Fatal> BaseDaemon: 7. DB::RowInputFormatWithDiagnosticInfo::getDiagnosticAndRawDataImpl(bool) @ 0x0000000012a5bc8b
2024.12.09 11:46:53.087827 [ 815 ] {} <Fatal> BaseDaemon: 8. DB::RowInputFormatWithDiagnosticInfo::getDiagnosticInfo() @ 0x0000000012a5be9d
2024.12.09 11:46:53.087852 [ 815 ] {} <Fatal> BaseDaemon: 9. DB::IRowInputFormat::read() @ 0x0000000012a2bb39
2024.12.09 11:46:53.087876 [ 815 ] {} <Fatal> BaseDaemon: 10. DB::IInputFormat::generate() @ 0x00000000129cc3d6
2024.12.09 11:46:53.087905 [ 815 ] {} <Fatal> BaseDaemon: 11. DB::ISource::tryGenerate() @ 0x00000000129a8b1b
2024.12.09 11:46:53.087931 [ 815 ] {} <Fatal> BaseDaemon: 12. DB::ISource::work() @ 0x00000000129a8827
2024.12.09 11:46:53.087972 [ 815 ] {} <Fatal> BaseDaemon: 13. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::ParallelParsingInputFormat::scheduleParserThreadForUnitWithNumber(unsigned long)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x0000000012b336e1
2024.12.09 11:46:53.087992 [ 815 ] {} <Fatal> BaseDaemon: 14. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::worker() @ 0x000000000d04eb8b
2024.12.09 11:46:53.088018 [ 815 ] {} <Fatal> BaseDaemon: 15. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000d0553a2
2024.12.09 11:46:53.088032 [ 815 ] {} <Fatal> BaseDaemon: 16. ThreadPoolImpl<std::thread>::ThreadFromThreadPool::worker() @ 0x000000000d04c0bf
2024.12.09 11:46:53.088048 [ 815 ] {} <Fatal> BaseDaemon: 17. void* std::__thread_proxy[abi:v15007]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void (ThreadPoolImpl<std::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::thread>::ThreadFromThreadPool*>>(void*) @ 0x000000000d052f5a
2024.12.09 11:46:53.088060 [ 815 ] {} <Fatal> BaseDaemon: 18. ? @ 0x00007f7f2ff5a609
2024.12.09 11:46:53.088089 [ 815 ] {} <Fatal> BaseDaemon: 19. ? @ 0x00007f7f2fe7f353
2024.12.09 11:46:53.292310 [ 815 ] {} <Fatal> BaseDaemon: Integrity check of the executable successfully passed (checksum: C2CF9CCDBE77F16F0CEC79B4A6228933)
2024.12.09 11:46:53.292695 [ 815 ] {} <Fatal> BaseDaemon: Report this error to https://github.com/ClickHouse/ClickHouse/issues
2024.12.09 11:46:53.293818 [ 815 ] {} <Fatal> BaseDaemon: Changed settings: max_threads = 8, max_query_size = 524288, connect_timeout_with_failover_ms = 1000, use_uncompressed_cache = false, enable_multiple_prewhere_read_steps = false, log_queries = true, http_receive_timeout = 1800., max_bytes_before_external_group_by = 27500000000, max_bytes_before_external_sort = 27500000000, max_memory_usage = 65000000000, max_memory_usage_for_user = 65000000000, external_storage_connect_timeout_sec = 2000, external_storage_rw_timeout_sec = 3000, allow_experimental_object_type = true, date_time_input_format = 'best_effort'
````

**Comment by Algunenano** [comment]
Could you please share a reproducer? Also please try the latest 24.10 or even master. It might be an already fixed bug related to deserialization on sparse columns directly (https://github.com/ClickHouse/ClickHouse/pull/72891 cc @CurtizJ )

**Comment by CurtizJ** [comment]
Yes, it is fixed in #72891.
